add Morakot autopilot

Rex-Taiphoon opened the pull request.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/defaults.parm:
```diff
@@ -0,0 +1,97 @@
+SERIAL0_BAUD,115
+SERIAL0_PROTOCOL,2
+
+# Serial1 setting (DJI / USART1)
```
remove section...only  add a define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_MSP_DisplayPort line in hwdef

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/defaults.parm:
```diff
@@ -0,0 +1,97 @@
+# Define ethernet settings
```
unless you have a really good reason, these should be left at inherent defaults and this section removed.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/hwdef.dat:
```diff
@@ -0,0 +1,163 @@
+
+# USB setup
+# USB_VENDOR 0x2DAE # ONLY FOR USE BY HEX! NOBODY ELSE
+# USB_PRODUCT 0x1058
```
remove

Hwurzburg reviewed (changes requested):
will need to re0review after changes

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/README.md:
```diff
@@ -0,0 +1,217 @@
+| 5   | FMU_TIM2_CH1_EXT| 3.3V         |
+| 6   | FMU_TIM2_CH2_EXT| 3.3V         |
+| 7   | FMU_TIM2_CH3_EXT| 3.3V         |
+| 8   | FMU_TIM2_CH4_EXT| 3.3V         |
```
sorry did not catch this the first time..
```suggestion
| 5   |  MOTOR5                  | 3.3V         |
| 6   |  MOTOR6                  | 3.3V         |
| 7   |  MOTOR7                  | 3.3V         |
| 8   |  MOTOR8                  | 3.3V         |
```

Rex-Taiphoon replied:
Sorry for the change, in our new FC revision we are reserving support for VTOL features.
Therefore, on the PWM2(ESC2) connector we have added additional PWM outputs, expanding it to five channels (PWM 5–9), as shown in the schematic below.
<img width="281" height="300" alt="image" src="https://github.com/user-attachments/assets/1a991230-0edc-4d44-9b37-66fade95dd11" />

I would like to confirm whether the README can be updated as follows:
#### ESC2 UART6 - 8 Pin JST-GH
| Pin | Signal Name     | Voltage      |
|-----|-----------------|--------------|
| 1   | VBAT IN         | 12V-33.6V    |
| 2   | GND             | 3.3V         |
| 3   | CURRENT         | 3.3V         |
| 4   | MOTOR/SERVO 5   | 3.3V         |
| 5   | MOTOR/SERVO 6   | 3.3V         |
| 6   | MOTOR/SERVO 7   | 3.3V         |
| 7   | MOTOR/SERVO 8   | 3.3V         |
| 8   | MOTOR/SERVO 9   | 3.3V         |

Hwurzburg commented:
added WIP since board changes are underway

Rex-Taiphoon commented:
Update summary:
1. Changed PWM output from 8 pins to 9 pins.
2. Moved ESC1 telemetry from UART7 to USART6 and removed ESC2 on USART6.
3. Removed unused or invalid comments.
4. Verified the default serial-port protocols in the latest version (defined in hwdef.dat).

Please help review these changes. Thank you!

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/hwdef.dat:
```diff
@@ -0,0 +1,163 @@
+PD6  USART2_RX USART2
+
+# USART3 (S.BUS)
+PD8  USART3_TX USART3 # Original
```
same here

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/hwdef.dat:
```diff
@@ -0,0 +1,163 @@
+
+IMU Invensensev3 SPI:imu1 ROTATION_ROLL_180_YAW_90
+
+define HAL_BARO_ALLOW_INIT_NO_BARO
```
you should remove this...it allows booting if the onboard baro is defective

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/README.md:
```diff
@@ -0,0 +1,215 @@
+| Pin | Signal Name     | Voltage      |
+|-----|-----------------|--------------|
+| 1   | VBAT IN         | 12V-33.6V    |
+| 2   | UART7_RX_EXT    | 3.3V         |
```
This should be UART6_RX by your hwdef and the wiring image changed

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/defaults.parm:
```diff
@@ -0,0 +1,21 @@
+# Serial7 setting (CSRF RC Input / UART8)
+SERIAL7_BAUD,57
+SERIAL7_OPTIONS,0
+SERIAL7_PROTOCOL,23
```
remove all this...duplicates what should be, and is, in the hwedf file

Rex-Taiphoon commented on libraries/AP_HAL_ChibiOS/hwdef/Morakot/README.md:
```diff
@@ -0,0 +1,217 @@
+- SERIAL3 -> USART3 (MAVLink2-Telem2(default)/S.BUS, DMA-enabled)
+- SERIAL4 -> UART5  (GPS, DMA-enabled)
+- SERIAL5->  USART6 (ESCTelemetry2, DMA-enabled)
+- SERIAL6 -> UART7  (MAVLink-Telem3 (default)/ ESCTelemetry1, DMA-enabled)
```
In our new morakot version, we have removed the ESC2 telemetry function previously assigned to R6, and moved the ESC1 telemetry function from R7 to R6, as shown in the schematic.
<img width="1198" height="278" alt="image" src="https://github.com/user-attachments/assets/58018a5b-53d0-44ec-86e9-51f789a35c2b" />

Below is our new uart mapping:
- SERIAL0 -> USB (MAVLink2)
- SERIAL1 -> USART1 (DisplayPort, DMA-enabled)
- SERIAL2 -> USART2 (MAVLink2-Telem1, DMA-enabled)
- SERIAL3 -> USART3 (MAVLink2-Telem2(default)/S.BUS, DMA-enabled)
- SERIAL4 -> UART5 (GPS, DMA-enabled)
- SERIAL5->  USART6 (ESCTelemetry1, DMA-enabled)
- SERIAL6 -> UART7 (MAVLink2-Telem3 (default), DMA-enabled)
- SERIAL7 -> UART8 (RC Input,  DMA-enabled)
